release-notes/swan-lake-GA-release-note.md

```diff
@@ -27,10 +27,238 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
 
 #### New Features
 
-#### Improvements
+##### Support to Refer to Parameters and Local Variables in Object Constructor Expressions
+
+Object constructor expressions can now refer to parameters and local variables. This feature is also available when using an object constructor expression with the `service` qualifier to create a service object. These references can be in field initializers and object methods including the `init` method.
+
+```ballerina
+type Person object {
+   string name;
+   string code;
+  
+   function getRegistrationId(string prefix) returns string;
+};
+ 
+function createPerson(string name, int id, string department) returns Person|error {
+   string codeId = check getCode(id);
+ 
+   Person person = object {
+       string name;
+       string code = codeId; // Refers to the `codeId` local variable.
+ 
+       function init() {
+           self.name = name; // Refers to the `name` parameter.
+       }
+ 
+       function getRegistrationId(string prefix) returns string {
+           string regId = department + ":" + self.code; // Refers to the `department` parameter.
+ 
+           if (prefix.length() > 0) {
+               return prefix + "-" + regId;
+           }
+ 
+           return regId;
+       }
+   };
+ 
+   return person;
+}
+ 
+function getCode(int id) returns string|error {
+   if id < 0 {
+       return error("Invalid ID");
+   }
+   return id.toString();
+}
+```
 
 #### Bug Fixes
 
+- A bug which resulted in constants being resolved with incorrect types in certain scenarios has been fixed. The type of constants is now the intersection of readonly and the singleton type containing just the shape of the value named by the constant.
```

Comment:
Is this correct? IMO reads like "because we fixed a bug constants are being resolved with incorrect types".

We've used this format previously also btw. E.g., https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-beta4

Reply:
Changed for consistency purposes as all bullet points to start with a verb in the past tense form. This is the practice followed in the other sections as well. (E.g., `Fixed`, `Added`, `Made`, etc.)

For example, see https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-beta6/#improvements.

Reply:
This is not what we had in previous lang PRs. 

Some of these are misleading now IMO.

https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-beta4/#change-in-expected-return-statements-in-a-function-with-an-optional-type-as-the-return-type says

```
Updated a function having an optional type that is not a subtype of error? as the return type to explicitly return a value.
```

whereas the original PR (https://github.com/ballerina-platform/ballerina-release/pull/1312/files#diff-38b0e1c5c10df6f38a8492a9042fab3d5be39ee28292b9502c5b986733eab3c6R329) had 

```
A function having an optional type that is not a subtype of `error?` as the return type is now expected to explicitly return a value.
```

These are not the same IMO. What we have in bio reads like we have updated a function. But what we wanted to say was the user should add these return statments!

Reply:
I will rephrase the GA bug fixes in the required form. But going forward can we make sure that significant changes such as these are suggested/introduced with the PRs the teams send itself, rather than the final PR sent for bio? I believe this is what we discussed also.

Looping in @anupama-pathirage also.

Reply:
Agreed with @MaryamZi here. Seems the meaning is completely different if we rephrase it as `Fixed a bug, which resulted in constants being resolve...`. To me it sounds like we fixed a bug and that resulted in another bug :) .

IMO, we should follow consistency practices only when it makes sense and only when the original meaning is not changing.

Reply:
+1 Will note these in the future. Thanks!